web/packages/shared/components/AccessRequests/NewRequest/RequestCheckout/RequestCheckout.tsx

```diff
@@ -921,6 +1191,14 @@ export type RequestCheckoutProps<T extends PendingListItem = PendingListItem> =
   {
     onClose(): void;
     toggleResource: (resource: T) => void;
+    toggleResources: (
+      resources: {
+        kind: T['kind'];
+        resourceId: T['id'];
+        resourceName: T['name'];
+      }[],
+      action?: 'add' | 'remove'
+    ) => void;
```

Comment:
Is there a specific reason we can't use `T` directly?

```suggestion
    toggleResources: (resources: T[], action?: 'add' | 'remove') => void;
```

Reply:
The issue with using just `T[]` here is we're creating new objects with only a subset of `T`'s properties (`{ resourceName, resourceId, kind }`), not passing instances of `T`